### PR TITLE
Added visual cueing when submitting forms

### PR DESCRIPTION
**Pull request by darinmclain**
Updated submit buttons with loaders upon submission, added success alerts

Changed ErrorProvider to become NotificationProvider, added success functionality and properties to that context.

### CODE REVIEW

File: src/components/Loading.tsx [hunk 1]
```diff
@@ -1,7 +1,22 @@
-export default function Loading() {
+type LoadingProps = {
+  isButton?: boolean;
```
Comment (markphelps) [inline src/components/Loading.tsx:2]: I dont think this is actually used anywhere else, the `Loading` component, so we could probably just make this only apply to buttons

File: src/components/flags/VariantForm.tsx [hunk 2]
```diff
@@ -44,6 +48,9 @@ export default function VariantForm(props: VariantFormProps) {
         handleSubmit(values)
           .then(() => {
             clearError();
+            setSuccess(
+              `Successfully ${submitPhrase.toLocaleLowerCase()}d variant.`
```
Comment (markphelps) [inline src/components/flags/VariantForm.tsx:52]: in some cases we end this message with a `.` and others we don't, like: https://github.com/flipt-io/flipt-ui/pull/52/files#diff-6fdc396f20dbeed901b28539b86f9ee4c7097a9f8aae89882a0c374dddfbb099R51

We should probably be consistent in the messaging/punctation, I'm just not sure which one I prefer (with the period or without). Thoughts?

Comment (markphelps) [reply src/components/flags/VariantForm.tsx:52]: I think I actually like without the `.`

File: src/components/SuccessNotification.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { Transition } from '@headlessui/react';
+import { CheckCircleIcon, XMarkIcon } from '@heroicons/react/20/solid';
+import { useSuccess } from '~/data/hooks/success';
+
+export default function SuccessNotification() {
+  const { success, setSuccess } = useSuccess();
+
+  return (
+    <Transition show={success !== null}>
```
Comment (markphelps) [inline src/components/SuccessNotification.tsx:9]: could we make the success notification clear itself after a short period? say 2-3 seconds? i think success notifications should be short-lived, while error notifications should persist until manually closed/acknowledged. wyt?

**Review by markphelps — CHANGES_REQUESTED** [review]
Looking good! Just a few minor things

File: src/components/flags/VariantForm.tsx (hunk 2 above)
Comment (darinmclain) [reply src/components/flags/VariantForm.tsx:52]: Removed the `.` from success messages

File: src/components/Loading.tsx (hunk 1 above)
Comment (darinmclain) [reply src/components/Loading.tsx:2]: Sounds good.  I wasn't sure if this was planned to be use elsewhere.  Removed the isButton logic/prop

File: src/components/SuccessNotification.tsx (hunk 3 above)
Comment (darinmclain) [reply src/components/SuccessNotification.tsx:9]: This is a good thought, I wasn't sure if this was a desired functionality.  Set timeout to be 3 seconds for now.

**Review by markphelps — APPROVED** [review]
lgtm! thank you!

**Pull request merged**